pandas/core/series.py

```diff
@@ -431,13 +431,21 @@ def nonzero(self):
 
     def put(self, *args, **kwargs):
         """
-        return a ndarray with the values put
+        Applies the `put` methods to its `values` attribute
+        and returns the resulting ndarray. Please refer to
+        `numpy.ndarray.put` for full documentation.
 
         See also
         --------
         numpy.ndarray.put
         """
-        self._values.put(*args, **kwargs)
+        try:
+            self._values.put(*args, **kwargs)
+        except AttributeError:
+            klass = type(self._values).__name__
```

Comment:
I guess that's fine for now. This is a funny method on categoricals. what are the ``values`` that are put? are they codes or categories? (and is the indexer codes or categories?). Look at how tricky ``__setitem__`` is. So if we don't need it then ok.